src/components/Sidebar/Sidebar.tsx

```diff
@@ -31,6 +36,12 @@ function SidebarComponent({
 }: Props) {
   const [active, activeSet] = useStateTransmorphic<string>(state);
   const [layoutStyles, displayStyles] = useLayoutStyles(style);
+  const sidebarRef = useRef<HTMLDivElement | null>(null);
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+
+  useEffect(() => {
```

Comment:
Since the sidebar now has a variable width to support different length text labels, The panel slideout effect needs an accompanying offset. The best way I could think to do this was to add an offset to the panel informed by the width of the sidebar (instead of a static `4.5rem`).

Reply:
Should `useLayoutEffect` be used here then?

Reply:
👍